### DISCUSSION TITLE
Error using df_pca$ind$coord

### DISCUSSION THREAD

**Discussion by smisani** [teamalgoritma/community#133] [Q&A] [answered]
Dear Algoritma team, 

I'm currently doing LBB  for Unsupervised Learning and I'm facing a problem in using df_pca$ind$coord. It keeps on showing error saying that:

> Error in pca_df$ind$coord[,1:8] : subscript out of bounds.

This puzzles me as my df has 16 Dimensions as shown below:
![image](https://user-images.githubusercontent.com/103245111/173268014-30ee3593-ea48-4529-8fc3-caa5a3c6965b.png)

What's even more puzzling is when I try to call pca_pdam$ind$coord, it has turned to 5 dimensions only as shown below:
![image](https://user-images.githubusercontent.com/103245111/173268156-d06394bd-2217-4ce0-a8d4-c8c84a744632.png)

Can anybody explain why this happens? 
I hope somebody can help me with this. Thanks~

Best,
SB

**Comment by VictorNugraha** [comment]
Hi, @smisani!

Thank you for the question.

First of all, I have a question for you. Is the PCA result saved into the same object? Since from your explanation and screenshot there are three different objects name (df_pca, pca_df & pca_pdam).

**Reply by smisani** [reply]
sorry for the confusion. 
I saved it as pca_pdam. df_pca/pca_df is just for a proxy for the purpose of asking. I hope this clarifies.

Best,
SB

**Reply by smisani** [reply]
the RMD file was already sent to mentor@algorit.ma. You may wanna check it out. 
Thanks in advance!

**Reply by onlyphantom** [reply]
What is the output of `dim(pca_pdam$ind$coord)` ? 

Quick tip: You can also `dput(pca_pdam)` or `dput(head(df))` and paste the output here so it can be easily reproduced -- provided that the data doesnt containing anything confidential :) `dput` deparses the object into a text representation that can be used to recreate said object, whether its a matrix, df, anything.

**Reply by smisani** [reply]
Hi @onlyphantom,

I'm afraid the data is confidential. But, I did share the dataset to mentor@algorit.ma provided that it should not be made public. So, appreciate it if you could get it from them if you happen to be from the mentor algoritma team.

By the way the output of `dim(pca_pdam$ind$coord)` is as shown below:

![image](https://user-images.githubusercontent.com/103245111/173324387-bf486f63-b8e0-4f2d-a965-bc48615abfd6.png)

Many thanks for your help in advance.

Best,
SB

**Comment by inytss** [comment, marked as answer]
Dear @smisani ,
I have checked your Rmd and dataset used on your LBB. As you get from the code `dim(pca_pdam$ind$coord)` you just have 5 dimensions means that you just can subset until 5 columns. So if you want to make the new data frame from the PCA result, you can use `pca_pdam$ind$coord[,1:5]`. 

Hope this will help you. If it does, please press the button of "Mark as answer".

**Reply by smisani** [reply]
Hi @inytss 

Thanks for the feedback, but if I use this code `pca_pdam$ind$coord[,1:5]`, will I be able to retain 80% of the information? Because when I looked at the code `summary(pca_pdam)` , it seems that I have to choose PC 1-8 to do so.

Appreciate your kind help in advance!

Cheers, 
SB

**Reply by inytss** [reply]
Dear @smisani ,
After we observe the code that you made in making PCA, it is as follows:
```
pca_pdam <- PCA(X = pdam_scale,
                  scale.unit = F,
                  graph = F)
```
where you don't set up the `ncp` parameter which describes how many PCs will be stored in the PCA result. If you don't set up the `ncp` parameter, the default value is 5 PCs that will be saved. Therefore, even though you have 17 variables used and in the output `pca_pdam$eig` displays 17 dimensions/PC, only 5 PCs will be stored.

So when you use `pca_pdam$ind$coord[,1:8]` there's an error occurs because only 5 PCs are stored. To overcome this, you can set the `ncp` parameter to as many variables as you use in making PCA. You can set it to 17 to create 17 PCs. Like this code:
```
pca_pdam <- PCA(X = pdam_scale,
                  scale.unit = F,
                  ncp=17,
                  graph = F)
```
 And if you want to save 80% information, you can use until PC8.

Hope this will help you.

**Reply by smisani** [reply]
This works~

Thanks so much!
